airbyte-webapp/src/components/common/ApiErrorBoundary/ApiErrorBoundary.tsx

```diff
@@ -72,6 +80,12 @@ class ApiErrorBoundaryComponent extends React.Component<
     }
   }
 
+  componentDidCatch(error: { message: string; status?: number; __type?: string }) {
+    if (isFormBuildError(error)) {
+      this.props.trackError(error);
```

Comment:
It would be nice if this could include a second argument to add some metadata about these errors, so they are easier to discover in datadog. I'm thinking of something like how errors are tracked in the useAuthenticator hook: https://github.com/airbytehq/airbyte/blob/2392acb845891f9c031b06feee9af2a2f975f1a7/airbyte-webapp/src/views/Connector/ConnectorForm/useAuthentication.tsx#L156-L159

If this could include an `id` like "formBuildError",  and also attach the connector ID to it, then we could see which connector the error occurred for. This seems important because I don't know how we would address a generic form build error that we see in datadog unless we know what connector it occurred for.